src/org/parosproxy/paros/security/CachedSslCertifificateServiceImpl.java

```diff
@@ -51,18 +52,24 @@ private CachedSslCertifificateServiceImpl() {
 	private Map<String, KeyStore> cache = new HashMap<>();
 
 	@Override
-	public synchronized KeyStore createCertForHost(String hostname)
+	public synchronized KeyStore createCertForHost(String hostname, InetAddress listeningAddress)
 			throws NoSuchAlgorithmException, InvalidKeyException,
 			CertificateException, NoSuchProviderException, SignatureException,
 			KeyStoreException, IOException, UnrecoverableKeyException {
-		if (this.cache.containsKey(hostname)) {
-			return this.cache.get(hostname);
+		String key = getKey(hostname, listeningAddress);
+		if (this.cache.containsKey(key)) {
+			return this.cache.get(key);
 		}
-		final KeyStore ks = delegate.createCertForHost(hostname);
-		this.cache.put(hostname, ks);
+		final KeyStore ks = delegate.createCertForHost(hostname, listeningAddress);
+		this.cache.put(key, ks);
 		return ks;
 	}
 
+	private String getKey(String hostname, InetAddress listeningAddress) {
+		String keyForCache = String.format("Host:%s;InetAddress:%s", hostname, listeningAddress.getHostAddress());

```

Comment:
It could return the value immediately (the method name already indicates what this is).